docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -231,6 +231,50 @@ var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), secretClient);
 
 This approach is useful when you would like to consolidate the dependency creation and share the client between multiple consuming classes.
 
+## Explore the extension methods in Azure Resource Manager SDKs
+
+For fundamental concepts of Azure Resource Manager SDKs, see [Key Concepts of Azure Resource Manager SDKs](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/resourcemanager/Azure.ResourceManager#key-concepts).
+
+In Azure Resource Manager SDKs, we design the clients (resources and collections) to have their hierarchical structures to emphasize their relationship between each other. To achieve that, extension methods are widely used because the parent resource of one resource might not exist in the same SDK library.
+
+For instance, in `Azure.ResourceManager.Compute` namespace, there is a resource `VirtualMachineResource` which stands for a virtual machine resource on Azure, and its parent is `ResourceGroupResource` in `Azure.ResourceManager` namespace, and we could get the `VirtualMachineCollection` on the `ResourceGroupResource` in this way:
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/ResourceManagerCodeStructure.cs" id="ParentOfVMIsRG" :::
+
+To make sure these extension methods are able to be mocked or unit tested, these methods are always implemented in this way:
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/ComputeExtensions.cs" id="HowExtensionMethodsAreImplemented" :::
+
+All the extension methods in Azure Resource Manager SDKs are implemented in the same pattern:
+
+1. There is a class holding the actual implementation of the method, we call it the "Mocking Extension" class. It is named in this pattern "{RP Name}{Resource Name}MockingExtension". In the above example, it is `ComputeResourceGroupMockingExtension`.
+1. The extension method has a private method constructing an instance of the "Mocking Extension" class by calling its `GetCachedClient` method.
+
+Both of the methods above are instance methods, therefore they are mockable and unit-testable. In this extension methods scenario, to create a test service client, you need to create two of them, one for the "Mocking Extension", and the other is the client that this extension method is extending.
+
+## [Non-library](#tab/csharp)
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/NonLibrary/MockComputeResourceGroupMockingExtension.cs" :::
+
+## [Moq](#tab/moq)
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/Moq/MockComputeResourceGroupMockingExtension_TestSnippets_Moq.cs" id="Moq_GetVirtualMachines" :::
```

Comment:
The `using` statements are displayed for the non-library approach. We should also display `using` statements for Moq and NSubstitute code samples.

Reply:
I would like to change, but how could I do that?

The usings and the code we would like to show stay at different levels on the same file - how could we put them together in the same snippet window?

Also in the `Understand service clients` section, we having usings in the `NonLibrary` part and no usings in `Moq` or `NSubstitute`. Should we change that part as well?

Reply:
> The usings and the code we would like to show stay at different levels on the same file - how could we put them together in the same snippet window?

In this example, remove the `id="Moq_GetVirtualMachines"` attribute from your `:::code` markup. Then you can remove the `// <Moq_GetVirtualMachines>` snippet markers from the code file. The contents of the entire file will display, which is the desired outcome.